src/devices/companion_wtr001.c

```diff
@@ -0,0 +1,153 @@
+| 13               | P: Parity to ensure count of set bits in data is odd.
+
+Temperature in Celcius = (bin2dec(bits 12,7,6,11,10,9,8) + ((bin2dec(bits 4,3,2,1,0) - 10) / 10 ) - 41
+
+Published range of device is -29.9C to 69.9C
+*/
+
+#include "decoder.h"
+
+#define MYDEVICE_BITLEN      14
+#define MYDEVICE_MINREPEATS  3
+
+static int companion_wtr001_decoder(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+
+    data_t *data;
+    int r; // a row index
+    uint8_t b[2];
+    float temperature;
+
+    // short pulse with long gap is 0, long pulse with short gap is 1.
+    // This is inverse of default OOK_PULSE_PWM behavior
+    bitbuffer_invert(bitbuffer);
+
+    r = bitbuffer_find_repeated_row(bitbuffer, MYDEVICE_MINREPEATS, MYDEVICE_BITLEN);
+    if (r < 0 || bitbuffer->bits_per_row[r] != 14) {
+        return 0;
+    }
+
+    bitbuffer_extract_bytes(bitbuffer, r, 0, b, 14);
+
+    // Make sure fixed bit in nibble 5 is not set
+    if ((b[0] & 0x04) == 0x04) {
+        if (decoder->verbose > 1) {
+            // fprintf(stderr, "companion_wtr001: Fixed Bit set (and it shouldn't be)\n");
+        }
+        return 0;
+    }
+
+    /* Parity check (must be ODD) */
+    if (!parity_bytes(b, 2)) {
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "companion_wtr001: parity check failed (should be ODD)\n");
+        }
+        return 0;
+    }
+
+    // Get the tenth of a degree C as nibbles 0,1,2,3,4 reversed, minus 0x0a
+    uint8_t temp_tenth_raw = reverse8(b[0] & 0xf8);
+   
+    if (temp_tenth_raw < 0x0a) {
+        // Value is too low
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "companion_wtr001: Temperature Degree Tenth too low (%d - 10 is less than 0\n", temp_tenth_raw);
+        }
+        return 0;
+    }
+    
+    if (temp_tenth_raw > 0x13) {
+        // Value is too high
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "companion_wtr001: Temperature Degree Tenth too high (%d - 10 is greater than 9\n", temp_tenth_raw);
+        }
+        return 0;
+    }
+
+    temp_tenth_raw -= 0x0a;
+    
+    uint8_t temp_whole_raw = reverse8(b[1]&0xf0) | reverse8(b[0]&0x03)>>2 | (b[1]&0x08)<<3;
+
+    if (temp_whole_raw < 11) {
+        // Value is too low (outside published specs)
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "companion_wtr001: Whole part of Temperature is too low (%d - 41 is less than -30)\n", temp_whole_raw);
+        }
+        return 0;
+    }
+    
+    if (temp_tenth_raw > 111) {
+        // Value is too high (outside published specs)
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "companion_wtr001: Whole part of Temperature is too high (%d - 41 is greater than 70)\n", temp_whole_raw);
+        }
+        return 0;
+    }
+   
+    temperature = (temp_whole_raw + (temp_tenth_raw * 0.1f)) - 41.0f;
+
+    data = data_make(
+            "model", "", DATA_STRING, _X("WTR001","Companion WTR001 Temperature Sensor"),
```

Comment:
No need for a deprecated "old model" name and _X macro. "Companion-WTR001" sounds right.